internal/services/catalog.go

```diff
@@ -98,6 +98,22 @@ func convertUnstructuredIntoCatalogService(unstructured unstructured.Unstructure
 		}
 	}
 
+	// if a secret was specified try to load the credentials from it
+	var repoUsername, repoPassword string
+	if catalogService.Spec.HelmRepo.Secret != "" {
+		authSecret, err := s.kubeClient.GetSecret(
+			context.Background(),
+			helmchart.Namespace(),
+			catalogService.Spec.HelmRepo.Secret,
+		)
+		if err != nil {
+			return nil, errors.Wrap(err, "finding helm repo auth secret")
```

Comment:
```suggestion
			return nil, errors.Wrap(err, "finding helm repo auth secret: "+catalogService.Spec.HelmRepo.Secret)
```
So that the user knows what secret to report to the operator as missing, along with the service in question.

Reply:
Make sense!